designs/senya/src/base.mjs

```diff
@@ -0,0 +1,219 @@
+import { waistlineHeight } from './shared.mjs'
+
+function infinitelyHigh(xCoordinate, Point) {
+  return new Point(xCoordinate, -100)
+}
+
+function infinitelyLow(xCoordinate, Point) {
+  return new Point(xCoordinate, 100)
+}
+
+function halfway(measurement, ease) {
+  return measurement / 2 + ease
+}
+
+function notHigherThan(a, b, Point) {
+  if (a.y < b.y) {
+    return new Point(a.x, b.y)
+  }
+  return a
+}
+
+function pointAtX(x, point, Point) {
+  return new Point(x, point.y)
+}
+
+function centerBack(measurements, options, Point) {
+  var hps = new Point(0, 0)
+  var neck = new Point(0, options.backNeckCutout * measurements.neck)
+  var waistline = new Point(
+    0,
+    waistlineHeight(
+      measurements.waistToUnderbust,
+      measurements.hpsToWaistBack,
+      options.lowerWaistline
+    )
+  )
+  return {
+    hps,
+    neck,
+    waistline,
+  }
+}
+
+function shoulderEase(measurements, options) {
+  return (measurements.shoulderToShoulder * options.shoulderEase) / 2
+}
+
+function neckPoint(measurements, options, Point) {
+  return new Point((measurements.neck * (1 + options.collarEase)) / options.collarFactor, 0)
+}
+
+function shoulder(hps, slope, width, ease, utils, Point) {
+  return utils.beamsIntersect(
+    hps,
+    hps.shift(slope * -1, 100),
+    infinitelyHigh(halfway(width, ease), Point),
+    infinitelyLow(halfway(width, ease), Point)
+  )
+}
+
+function armhole(shoulderY, measurements, options, Point) {
+  return new Point(
+    0,
+    shoulderY + measurements.biceps * (1 + options.bicepsEase) * options.armholeDepthFactor
+  )
+}
+
+function adjustShoulderSlope(shoulder, centerBackHps, slopeReduction) {
+  shoulder.y -= (shoulder.y - centerBackHps.y) * slopeReduction
+}
+
+function halfwidth(chest, chestEase) {
+  return (chest * (1 + chestEase)) / 4
+}
+
+function fitCollar(neckCircumverence, collarEase, neck, neckCp2, cbNeck, utils, Point, Path) {
+  let cf = neck.rotate(-90, new Point(0, 0))
+  let target = neckCircumverence * (1 + collarEase)
+  let delta = 0
+  let run = 0
+  do {
+    run++
+    cf = cf.shift(90, delta / 3)
+    // frontNeckCpEdge aux
+    let frontNeckCpEdge = utils.beamsIntersect(neck, neckCp2, cf, new Point(20, cf.y))
+    // cfNeckCp1 needed
+    var cfCp1 = cf.shiftFractionTowards(frontNeckCpEdge, 0.55)
```

Comment:
No `var`s please, stick to `const` and `let.

Declare your `let`s outside the loop, then assign them in the loop:
```mjs
let cfCp1, cp2Front
do {
    run++
    // ...
    cfCp1 = cf.shiftFractionTowards(frontNeckCpEdge, 0.55)
    cp2Front = neck.shiftFractionTowards(frontNeckCpEdge, 0.65)
}
```

Reply:
Hah, interesting style, definitely makes sense for readability, changed.